javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/TypeExtractor.java

```diff
@@ -360,7 +363,7 @@ public ResolvedType visit(SuperExpr node, Boolean solveLambdas) {
                 if (resolvedTypeName.isInterface()) {
                     return new ReferenceTypeImpl(resolvedTypeName.asInterface(), typeSolver);
                 } else if (resolvedTypeName.isClass()) {
-                    return resolvedTypeName.asClass().getSuperClass();
+                    return resolvedTypeName.asClass().getSuperClass().orElseThrow(() -> new RuntimeException("super class unexpectedly empty"));
```

Comment:
There are many cases throughout this PR which make use of `orElseThrow` -- I consider this to be okay to include as "default" given that the current behaviour in many cases is a value of `null` (and subsequent `NullPointerException` runtime errors further down the line.